lib/ansible/modules/cloud/amazon/ec2_group.py

```diff
@@ -578,17 +615,9 @@ def main():
         # Finally, remove anything left in the groupRules -- these will be defunct rules
         if purge_rules_egress:
             for (rule, grant) in groupRules.values():
-                grantGroup = None
-                if grant.group_id:
-                    grantGroup = groups[grant.group_id].id
+                ipPermissions = createRevokeIpPermission(grant, rule)
                 if not module.check_mode:
-                    ec2.revoke_security_group_egress(
-                        group_id=group.id,
-                        ip_protocol=rule.ip_protocol,
-                        from_port=rule.from_port,
-                        to_port=rule.to_port,
-                        src_group_id=grantGroup,
-                        cidr_ip=grant.cidr_ip)
+                    group.revoke_egress(GroupId=group.id, IpPermissions=[ipPermissions])
```

Comment:
Using http://boto3.readthedocs.io/en/latest/reference/services/ec2.html#EC2.Client.revoke_security_group_egress is more usual for most of our boto3 module (and more similar to the existing code). This is just one example, the same applies for other actions on the `SecurityGroup` class